### PR TITLE
xfce.xfmpc: init at 0.3.1-2024-05-29

### PR DESCRIPTION
**Pull request by doronbehar**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/xf/xfmpc/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    # From some reason, upstream conditioned this maintainer mode with the
+    # search of the vala compiler in `configure.ac`. Without `VALAC` set in the
+    # makefiles, the build fails with a wierd and hard to understand error.
+    "--enable-maintainer-mode"
+  ];
+
+  meta = with lib; {
+    description = "MPD client written in GTK";
+    homepage = "https://gitlab.xfce.org/apps/xfmpc";
+    changelog = "https://gitlab.xfce.org/apps/xfmpc/-/blob/${src.rev}/NEWS";
```
Comment (Noodlez1232) [inline pkgs/by-name/xf/xfmpc/package.nix:48]: You gotta use `finalAttrs.src`

**Review by Noodlez1232 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/xf/xfmpc/package.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/by-name/xf/xfmpc/package.nix]: > You gotta use `finalAttrs.src`. It doesn't build otherwise.

Thanks!

**Comment by doronbehar** [comment]
@ofborg eval

File: pkgs/desktops/xfce/applications/xfmpc/default.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  src = fetchFromGitLab {
+    domain = "gitlab.xfce.org";
+    owner = "apps";
+    repo = "xfmpc";
+    rev = "cf40dffec6e9b80abb1f1aa6d7dceef4790173dc";
+    hash = "sha256-moCWSLGBJuWM4/lRJi6D3w38iJeCntLo3Vl/eVfu7lw=";
+  };
+  nativeBuildInputs = [
+    autoreconfHook
+    xfce4-dev-tools
```
Comment (bobby285271) [inline pkgs/desktops/xfce/applications/xfmpc/default.nix:29]: In `xdtAutogenPhase` I would [expect](https://gitlab.xfce.org/xfce/xfce4-dev-tools/-/blob/4bd037dbe567d9161e7a65108e31f955194a570b/scripts/xdt-autogen.in#L437) xdt-autogen to trigger autoreconf, do we have to keep `autoreconfHook` here triggering autoreconf twice?

File: pkgs/desktops/xfce/applications/xfmpc/default.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    pkg-config
+    vala
+    # Needed both here and in buildInputs for cross compilation to work
+    libxfce4ui
+    libxfce4util
```
Comment (bobby285271) [inline pkgs/desktops/xfce/applications/xfmpc/default.nix:34]: Ugh, it looks like the cross `libxfce4{ui,util}` build failed to find native `vapigen` and thus not building the `*.vapi` files.

File: pkgs/desktops/xfce/applications/xfmpc/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    gtk3
+    glib
+    libxfce4ui
+    libmpd
+  ];
+  configureFlags = [
+    # From some reason, upstream conditioned this maintainer mode with the
+    # search of the vala compiler in `configure.ac`. Without `VALAC` set in the
+    # makefiles, the build fails with a wierd and hard to understand error.
+    "--enable-maintainer-mode"
```
Comment (bobby285271) [inline pkgs/desktops/xfce/applications/xfmpc/default.nix:46]: `mkXfceDerivation` is always passing this:

https://github.com/NixOS/nixpkgs/blob/b57526ee5cf47504b759cda0b3655a73b40e9a61/pkgs/desktops/xfce/mkXfceDerivation.nix#L30

[Release tarball](https://archive.xfce.org/src/apps/xfmpc/) dists generated C files so I think neither the maintainer mode nor valac will be needed there. It looks like devs  [don't quite understand](https://matrix.to/#/!lhGQfSoqXljRjKiRMt:matrix.org/$7cHH2oMaUcMnkNYZ8Zc5cH8gwyR2_0OWxDQxBsysjTM?via=matrix.org&via=tarricone.org&via=tchncs.de) why we are not using release tarballs 🙃

File: pkgs/desktops/xfce/applications/xfmpc/default.nix [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  ];
+
+  meta = with lib; {
+    description = "MPD client written in GTK";
+    homepage = "https://docs.xfce.org/apps/xfmpc/start";
+    changelog = "https://gitlab.xfce.org/apps/xfmpc/-/blob/${finalAttrs.src.rev}/NEWS";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ doronbehar ];
```
Comment (bobby285271) [inline pkgs/desktops/xfce/applications/xfmpc/default.nix:54]: To also show up on my repology [profile](https://repology.org/maintainer/rjl931189261@126.com) 😂 

```suggestion
    maintainers = with maintainers; [ doronbehar ] ++ teams.xfce.members;
```

Comment (doronbehar) [reply pkgs/desktops/xfce/applications/xfmpc/default.nix]: > To also show up on my repology

Haha, welcome to the maintainance team :).

File: pkgs/desktops/xfce/applications/xfmpc/default.nix (hunk 4 above)
Comment (doronbehar) [reply pkgs/desktops/xfce/applications/xfmpc/default.nix]: > It looks like devs [don't quite understand](https://matrix.to/#/!lhGQfSoqXljRjKiRMt:matrix.org/$7cHH2oMaUcMnkNYZ8Zc5cH8gwyR2_0OWxDQxBsysjTM?via=matrix.org&via=tarricone.org&via=tchncs.de) why we are not using release tarballs 🙃

Couldn't open that link.. But I understand it's not super important to understand it.

Comment (doronbehar) [reply pkgs/desktops/xfce/applications/xfmpc/default.nix]: > `mkXfceDerivation` is always passing this:

In general it's a good idea to use it!

File: pkgs/desktops/xfce/applications/xfmpc/default.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/desktops/xfce/applications/xfmpc/default.nix]: > In `xdtAutogenPhase` I would [expect](https://gitlab.xfce.org/xfce/xfce4-dev-tools/-/blob/4bd037dbe567d9161e7a65108e31f955194a570b/scripts/xdt-autogen.in#L437) xdt-autogen to trigger autoreconf, do we have to keep `autoreconfHook` here triggering autoreconf twice?

You were right! Ended up using only `xdtAutogenPhase`.

**Review by bobby285271 — APPROVED** [review]
Works for me.

**Pull request merged**